src/dataEntities/appointment.ts

```diff
@@ -271,9 +283,21 @@ export class Appointment {
      * Validate property values on the appointment
      * @param log Logger to be used in case of failures
      */
-    public async validate(log: Logger = logger) {
+    public async validate(
+        blockCache: ReadOnlyBlockCache<IBlockStub>,
+        pisaContractAddress: string,
+        log: Logger = logger
+    ) {
         if (this.paymentHash.toLowerCase() !== Appointment.FreeHash) throw new PublicDataValidationError("Invalid payment hash."); // prettier-ignore
 
+        // the start block must be close to the current block +/- 5
+        const currentHead = blockCache.head.number;
+        if(this.startBlock < (currentHead - 5)) throw new PublicDataValidationError(`Start block too low. Start block must be within 5 blocks of the current block ${currentHead}.`); // prettier-ignore
+        if(this.startBlock > (currentHead + 5)) throw new PublicDataValidationError(`Start block too high. Start block must be within 5 blocks of the current block ${currentHead}.`); // prettier-ignore
+        if((this.endBlock - this.startBlock) > 60000) throw new PublicDataValidationError(`Appointment duration too great. Maximum duration between start and block is 60000.`); // prettier-ignore
```

Comment:
typo, "between start and block" => "between start and end block"

Reply:
Done